org/postgresql/jdbc2/TypeInfoCache.java

```diff
@@ -150,9 +153,14 @@ public synchronized void addCoreType(String pgTypeName, Integer oid, Integer sql
         Character delim = ',';
         _arrayOidToDelimiter.put(oid, delim);
 
-        String pgArrayTypeName = "_" + pgTypeName;
```

Comment:
This `_ => []` change does break things, see https://github.com/pgjdbc/pgjdbc/pull/592
@zapov , can you please clarify what was the intention behind `[]` change? Was it just for "simplified isArray by mere `endsWith("[]")`"?

As backend reports underscores in `information_schema`, I think we'd better stick with underscore kind of type names.

// cc @isopov